enterprise_data/api/v1/serializers.py

```diff
@@ -63,35 +59,38 @@ class Meta:
         model = EnterpriseOffer
         fields = '__all__'
 
-    def validate_offer_id(self, value) -> str:
+    def to_internal_value(self, data):
         """
+        Convert the incoming data offer_id field to a format that can be stored in the db.
+
         For a given offer_id string from the requester, determine the best representation to use for db storage.
 
         Raises serializers.ValidationError:
             If the given string is not exclusively numeric characters, but also does not parse as a UUID (either because
             it has the wrong length, incorrect dashes, or some other reason).
         """
-        LOGGER.info('Validating offer ID: %s', value)
-        if len(value) < 10 and isinstance(value, int):
-            LOGGER.info('Validated offer ID is int: %s', value)
-            return value
-
-        elif isinstance(value, str) and len(value) == 32:
-            LOGGER.info('Validated offer ID is string/UUID: %s', value)
-            return value
-        else:
-            raise serializers.ValidationError("requested offer_id neither a valid integer nor UUID.")
-
-    def to_internal_value(self, data):
-        """
-        Convert the incoming data offer_id field to a format that can be stored in the db.
-        """
-        LOGGER.info('Converting offer ID to internal value to_internal_value: %s', data)
-        ret = super().to_representation(data)
-        if isinstance(ret['offer_id'], str):
-            ret['offer_id'] = ret['offer_id'].replace('-', '')
-        LOGGER.info('Converted offer ID to internal value to_internal_value: %s', ret)
-        return ret
+        ret = super().to_internal_value(data)
+        if ret['offer_id'] is None or ret['offer_id'] == '':
+            raise serializers.ValidationError("requested offer_id is None.")
+
+        if isinstance(ret['offer_id'], str) and len(ret['offer_id']) == 36:
+            offer_id = ret['offer_id'].replace('-', '')
+            # There should only be 4 dashes in the UUID, making the length 32 after removal
+            if len(offer_id) == 32:
+                ret['offer_id'] = offer_id
+                return ret
+
+            else:
+                raise serializers.ValidationError("requested offer_id neither a valid integer nor UUID.")
+
+        if len(ret['offer_id']) < 10:  # All ecommerce offer_ids are at < 1 million.
+            try:
+                ret['offer_id'] = int(ret['offer_id'])
+                return ret
```

Comment:
this is causing an attempt to store an integer into the database.  It'll work as intended, but cleaner to just use int() as a test:
```suggestion
                int(ret['offer_id'])
                return ret
```